packages/terra-application-links/src/tabs/_TabMenuDisplay.jsx

```diff
@@ -38,50 +43,88 @@ class TabMenuDisplay extends React.Component {
   constructor(props, context) {
     super(props, context);
 
-    this.update = this.update.bind(this);
+    this.handleKeyDown = this.handleKeyDown.bind(this);
+    this.handleKeyUp = this.handleKeyUp.bind(this);
+    this.handleOnBlur = this.handleOnBlur.bind(this);
+
+    this.state = { focused: false };
+  }
 
-    this.hasAnimated = false;
+  handleOnBlur() {
+    if (!this.props.popup) {
+      this.setState({ focused: false });
+    }
   }
 
-  componentDidMount() {
-    window.requestAnimationFrame(this.update);
+  handleKeyDown(event) {
+    // Add active state to FF browsers
+    if (event.nativeEvent.keyCode === TabUtils.KEYCODES.SPACE) {
+      this.setState({ active: true });
+    }
+
+    // Add focus styles for keyboard navigation
+    if (event.nativeEvent.keyCode === TabUtils.KEYCODES.SPACE || event.nativeEvent.keyCode === TabUtils.KEYCODES.ENTER) {
+      this.setState({ focused: true });
+
+      event.preventDefault();
+      if (this.props.onKeyDown) {
+        this.props.onKeyDown(event);
+      }
+    }
   }
 
-  update() {
-    if (this.props.isSelected && this.contentNode && !this.hasAnimated) {
-      this.contentNode.setAttribute('aria-current', 'true');
-      this.hasAnimated = true;
+  handleKeyUp(event) {
+    // Remove active state from FF broswers
+    if (event.nativeEvent.keyCode === TabUtils.KEYCODES.SPACE) {
+      this.setState({ active: false });
+    }
+
+    // Apply focus styles for keyboard navigation
+    if (event.nativeEvent.keyCode === TabUtils.KEYCODES.TAB) {
+      this.setState({ focused: true });
     }
   }
 
   render() {
     const {
       isHidden,
       isSelected,
+      onKeyDown,
       popup,
       refCallback,
       text,
       ...customProps
     } = this.props;
 
-    const attributes = {};
-    if (isSelected && this.hasAnimated) {
-      attributes['aria-current'] = true;
-    }
+    const displayClassNames = cx([
+      'tab-menu-display',
+      { 'is-hidden': isHidden },
+      { 'is-active': this.state.active },
+      { 'is-focused': this.state.focused },
+      customProps.className,
+    ]);
+    const attributes = { 'aria-current': isSelected };
 
     return (
+      /* eslint-disable jsx-a11y/no-static-element-interactions */
```

Comment:
We should get rid of this a11y lint. You should just need to add an interactive role to the div.

Reply:
Removing that rule still fails to lint, it has a role="tab", but even switching it to role="button" it still fails. Is there a different role i should be setting?